fdbserver/storageserver.actor.cpp

```diff
@@ -5519,25 +5526,12 @@ ACTOR Future<Void> getMappedKeyValuesQ(StorageServer* data, GetMappedKeyValuesRe
 				//                ASSERT(r.data.size() <= std::abs(req.limit));
 			}
 
-			// For performance concerns, the cost of a range read is billed to the start key and end key of the range.
-			int64_t totalByteSize = 0;
-			for (int i = 0; i < r.data.size(); i++) {
-				totalByteSize += r.data[i].expectedSize();
-			}
-			if (totalByteSize > 0 && SERVER_KNOBS->READ_SAMPLING_ENABLED) {
-				int64_t bytesReadPerKSecond = std::max(totalByteSize, SERVER_KNOBS->EMPTY_READ_PENALTY) / 2;
-				data->metrics.notifyBytesReadPerKSecond(addPrefix(r.data[0].key, req.tenantInfo.prefix, req.arena),
-				                                        bytesReadPerKSecond);
-				data->metrics.notifyBytesReadPerKSecond(
-				    addPrefix(r.data[r.data.size() - 1].key, req.tenantInfo.prefix, req.arena), bytesReadPerKSecond);
-			}
-
```

Comment:
If the secondary queries are served locally, maybe it makes sense to keep these? It seems if the secondary queries are served from other storage servers, the number of bytes read doesn't make much sense here.

Reply:
if it is served locally, it will run [this line ](https://github.com/apple/foundationdb/blob/main/fdbserver/storageserver.actor.cpp#L4417), and [here ](https://github.com/apple/foundationdb/blob/main/fdbserver/storageserver.actor.cpp#L4331-L4341) it calculate the bytesReadPerKSecond, because it has the identical code as here.